refactor: update benchmarks and tests in `math/base/special/acosh`

anandkaranubc opened the pull request:
Resolves none.

## Description

> What is the purpose of this pull request?

This pull request:

- Refactors random number generation in JS benchmarks for `math/base/special/acosh`
- Replaces `randu()` with `uniform()` from `@stdlib/random/array/uniform` for cleaner and more consistent code.
- Moves the random number generation outside the benchmarking loops.
- Updates the test messages to follow code conventions.

## Related Issues

> Does this pull request have any related issues?

This pull request:

- resolves none

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

anandkaranubc committed:
refactor: benchmarks and test messages

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: missing_dependencies
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

anandkaranubc commented:
The non-exact tests in the test files were also removed because they were not being executed.  

A follow-up question regarding this refactoring: The refactoring is mainly being done for C/C++ or JS benchmarks. Is there a need to apply it to the Julia, R, or Python benchmarks as well? I believe so..?

kgryte commented:
> Is there a need to apply it to the Julia, R, or Python benchmarks as well?

@anandkaranubc Technically, yes, but this is very low priority, so I think we can punt to the future.

kgryte commented on lib/node_modules/@stdlib/math/base/special/acosh/test/test.js:
```diff
@@ -59,19 +56,13 @@ tape( 'the function computes the hyperbolic arccosine on the interval [1.0,3.0]'
 		y = acosh( x[i] );
 		if ( y === expected[ i ] ) {
 			t.equal( y, expected[ i ], 'x: '+x[i]+'. E: '+expected[i] );
-		} else {
```
@anandkaranubc These changes should be reverted or you need to remove the `if` branch. Right now, if we were to update the test fixtures and the upstream values were to change, we would not catch that, as you now only assert when the values are equal. In this case, the approximate branch was included as a fallback in case values were to ever differ.

anandkaranubc replied:
Sure, I will revert those. Is this different from what's being done in `stats/base/dists/*`. It was pointed out to me by @Neerajpathak07 in one of my PRs. [Discussion](https://github.com/stdlib-js/stdlib/pull/3974#discussion_r1950126777)

anandkaranubc replied:
Done!

kgryte replied:
@anandkaranubc That comment applies to naive calculations. See the C implementation: https://github.com/stdlib-js/stdlib/pull/3974/files#diff-88bd0b2a5a78d2d32485c3217f1784e428ebf5cae61ec45eb5a1b872f400e0d7. In that implementation, there is no computation. Here, for `acosh`, we would, however, expect some approximate results due to floating-point rounding error.

anandkaranubc replied:
Ohhhh, I see! Sorry, I overlooked that. Thanks for the clarification.

kgryte commented on lib/node_modules/@stdlib/math/base/special/acosh/test/test.native.js:
```diff
@@ -68,19 +65,13 @@ tape( 'the function computes the hyperbolic arccosine on the interval [1.0,3.0]'
 		if ( y === expected[ i ] ) {
 			t.equal( y, expected[ i ], 'x: '+x[i]+'. E: '+expected[i] );
-		} else {
-			delta = abs( y - expected[i] );
```
Same comment.

anandkaranubc committed:
test: revert the changes to include else block

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: passed
---

kgryte reviewed (approved):
LGTM
